handler/src/main/java/io/netty/handler/ssl/SniHandler.java

```diff
@@ -313,31 +333,58 @@ public void operationComplete(Future<SslContext> future) throws Exception {
     }
 
     /**
-     * The default implementation will simply call {@link AsyncMapping#map(Object, Promise)} but
-     * users can override this method to implement custom behavior.
+     * The default implementation will simply call {@link SslContextMapping#map(String, Promise)} but users can
+     * override this method to implement custom behavior.
+     *
+     * NOTE: This method returns a {@code Future<?>} to accommodate the old and new way of doing things without
+     * braking the API. Users who override this method are advised to change their return value from
+     * {@code Future<SslContext>} to {@code Future<Supplier<? extends SslContext>>}.
      *
-     * @see AsyncMapping#map(Object, Promise)
+     * @see SslContextMapping#map(String, Promise)
+     * @deprecated the return value will at some point change from {@code Future<SslContext>} to
+     *   {@code Future<Supplier<? extends SslContext>>}.
      */
-    protected Future<SslContext> lookup(ChannelHandlerContext ctx, String hostname) throws Exception {
-        return mapping.map(hostname, ctx.executor().<SslContext>newPromise());
+    @Deprecated
+    protected Future<?> lookup(ChannelHandlerContext ctx, String hostname) throws Exception {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove usage of generic wildcard type. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1452)